packages/flutter/lib/src/widgets/scroll_metrics.dart

```diff
@@ -98,12 +100,23 @@ abstract class ScrollMetrics {
   /// The quantity of visible content.
   ///
   /// If [extentBefore] and [extentAfter] are non-zero, then this is typically
-  /// the height of the viewport. It could be less if there is less content
-  /// visible than the size of the viewport.
+  /// the height of the viewport. The value will always be greater than or equal
+  /// to the size of the viewport.
+  ///
+  /// The implementation assumes [maxScrollExtent] is greater than or equal to
+  /// [minScrollExtent] and thus the value will always be greater than or equal
+  /// to [viewportDimension].
+  ///
+  /// See also:
+  ///
+  /// * [ViewportOffset.applyContentDimensions]
   double get extentInside {
+    assert(maxScrollExtent != null);
+    assert(minScrollExtent != null);
+    assert(minScrollExtent <= maxScrollExtent);
     return math.min(pixels, maxScrollExtent) -
            math.max(pixels, minScrollExtent) +
-           math.min(viewportDimension, maxScrollExtent - minScrollExtent);
```

Comment:
What was this for again? I think we're flipping the previous definition now.

Reply:
This is to fix #31430. According to the documentation of [ViewportOffset.applyContentDimensions](https://docs.flutter.io/flutter/rendering/ViewportOffset/applyContentDimensions.html):

> The maximum scroll extent has the viewport dimension subtracted from it. For instance, if there is 100.0 pixels of scrollable content, and the viewport is 80.0 pixels high, then the minimum scroll extent will typically be 0.0 and the maximum scroll extent will typically be 20.0, because there's only 20.0 pixels of actual scroll slack.

So `maxScrollExtent - minScrollExtent` is not the size of the content, `maxScrollExtent - minScrollExtent + viewportDimension` is.